### PR TITLE
[formrecognizer] Adding sync and async samples for ID documents

### PR DESCRIPTION
**Pull request by catalinaperalta**
Addresses #16683

### CODE REVIEW

File: sdk/formrecognizer/azure-ai-formrecognizer/samples/async_samples/sample_recognize_id_documents_async.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+import asyncio
+
+
+class RecognizeIdDocumentsSampleAsync(object):
+
+    async def recognize_id_document(self):
+        path_to_sample_forms = os.path.abspath(os.path.join(os.path.abspath(__file__),
+                                                            "..", "./../sample_forms/id_documents/license.jpg"))
+
+        # [START recognize_id_document]
```
Comment (kristapratico) [inline sdk/formrecognizer/azure-ai-formrecognizer/samples/async_samples/sample_recognize_id_documents_async.py:36]: to match the end clause
```suggestion
        # [START recognize_id_documents]
```

File: sdk/formrecognizer/azure-ai-formrecognizer/samples/async_samples/sample_recognize_id_documents_async.py [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+                sex = id_document.fields.get("Sex")
+                if sex:
+                    print("Sex: {} has confidence: {}".format(sex.value_data.text, sex.confidence))
+                address = id_document.fields.get("Address")
+                if address:
+                    print("Address: {} has confidence: {}".format(address.value, address.confidence))
+                # FIXME: uncomment this
+                # country = id_document.fields.get("Country")
+                # if country:
+                #     print("Country: {} has confidence: {}".format(country.value, country.confidence))
```
Comment (kristapratico) [inline sdk/formrecognizer/azure-ai-formrecognizer/samples/async_samples/sample_recognize_id_documents_async.py:78]: is it not finding "Country"?

File: sdk/formrecognizer/azure-ai-formrecognizer/samples/async_samples/sample_recognize_id_documents_async.py (hunk 1 above)
Comment (catalinaperalta) [reply sdk/formrecognizer/azure-ai-formrecognizer/samples/async_samples/sample_recognize_id_documents_async.py:36]: I'd been going back and forth about having the `s` at the end. Do you think I should rename the functions to have an `s` at the end?

Comment (catalinaperalta) [reply sdk/formrecognizer/azure-ai-formrecognizer/samples/async_samples/sample_recognize_id_documents_async.py]: Fixed. I decided to add the `s` to the function as well.

File: sdk/formrecognizer/azure-ai-formrecognizer/samples/async_samples/sample_recognize_id_documents_async.py (hunk 2 above)
Comment (kristapratico) [reply sdk/formrecognizer/azure-ai-formrecognizer/samples/async_samples/sample_recognize_id_documents_async.py:78]: Oh okay let's leave it commented out, I thought it wasn't finding the field at all

**Pull request merged**